### PR TITLE
Update sdfstudio_dataparser.py

### PR DESCRIPTION
**Pull request by qsun1**
Explicitly split training set and validation set, when there is no split provided. The default validation split will be the indices that can be divided exactly by an integer, i.e. `self.config.skip_every_for_val_split`. 

Current implementation might cause confusion as validation set is contained by the training set.

Thanks for your wonderful project and your effort.

### CODE REVIEW

**Comment by niujinshuchong** [comment]
@qsun1 Thanks for the PR. We use all the images for training to make it consistent with previous work. For example, NeuS and VolSDF use all DTU images for training. Could you add it as an option so we could exclude testing images when it's specified from the training command.

**Comment by qsun1** [comment]
@niujinshuchong Thanks for your advise. I've add the ```train_val_no_overlap``` option for the training command.

I test the code in my device:
```bash
MODEL="neus" 
INDOOR="True" 
DATASET="replica-scan1"
PATH_DATA="../Replica/scan1"
train_val_no_overlap="True" # options

ns-train ${MODEL} --pipeline.model.sdf-field.inside-outside ${INDOOR} --vis tensorboard --experiment-name ${MODEL}-${DATASET} sdfstudio-data --data ${PATH_DATA} --skip_every_for_val_split 10 --train_val_no_overlap ${train_val_no_overlap}
```
The size of training set would be 90 instead of 100 (number of all images in the dataset).
```python
Setting up training dataset...
Caching all 90 images.
Setting up evaluation dataset...
Caching all 10 images.
```

File: nerfstudio/data/dataparsers/sdfstudio_dataparser.py [hunk 1]
```diff
@@ -181,6 +183,10 @@ def _generate_dataparser_outputs(self, split="train"):  # pylint: disable=unused
         # subsample to avoid out-of-memory for validation set
         if split != "train" and self.config.skip_every_for_val_split >= 1:
             indices = indices[:: self.config.skip_every_for_val_split]
+        else:
+            # if you use this option, training set should not contain any image in validation set
+            if self.config.train_val_no_split:
```
[the remark this answers is not in the document]
Comment (qsun1) [reply nerfstudio/data/dataparsers/sdfstudio_dataparser.py]: sorry for my typo, fixed in the new commit
